apps/design/backend/src/worker/worker.ts

```diff
@@ -1,45 +1,46 @@
 import { extractErrorMessage, sleep } from '@votingworks/basics';
 
-import { BackgroundTask, Store } from '../store';
+import { Workspace } from '../workspace';
+import { processBackgroundTask } from './tasks';
 
-async function processBackgroundTask(
-  _store: Store, // eslint-disable-line @typescript-eslint/no-unused-vars
-  _task: BackgroundTask // eslint-disable-line @typescript-eslint/no-unused-vars
-): Promise<void> {
-  // Simulate a long-running operation
-  // TODO(arsalan): Implement actual processing logic
-  await sleep(3000);
-}
+export async function processNextBackgroundTaskIfAny({
+  log = console.log, // eslint-disable-line no-console
+  workspace,
+}: {
+  log?: (text: string) => void;
+  workspace: Workspace;
+}): Promise<void> {
+  const { store } = workspace;
 
-async function processBackgroundTasks(store: Store): Promise<void> {
-  // eslint-disable-next-line no-constant-condition
-  while (true) {
-    const nextTask = store.getOldestQueuedBackgroundTask();
+  const nextTask = store.getOldestQueuedBackgroundTask();
 
-    if (!nextTask) {
-      await sleep(1000);
-      continue;
-    }
+  if (!nextTask) {
+    await sleep(1000);
```

Comment:
This feels like something that should happen outside this function, i.e. in the main loop. Return whether a job was processed and sleep if not?

Reply:
Agreed: https://github.com/votingworks/vxsuite/pull/4462/commits/9b88b0bee4f87ddee6a7cf6efe4a9b722b3b9263